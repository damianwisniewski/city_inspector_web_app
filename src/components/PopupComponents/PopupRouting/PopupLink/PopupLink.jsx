import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { PopupContext } from '../PopupRouter/PopupRouter'

class PopupLink extends Component {
	static contextType = PopupContext

	render() {
		const { children, to, activeClassName, className, ...props } = this.props
		const { handleOpenPopup, componentName } = this.context

		return (
			<button
				{...props}
				data-to={to}
				onClick={handleOpenPopup}
				className={`${className} ${activeClassName && to === componentName ? activeClassName : ''}`}
			>
				{children}
			</button>
		)
	}
}

PopupLink.propTypes = {
	children: PropTypes.node,
	to: PropTypes.string.isRequired,
}

export default PopupLink
