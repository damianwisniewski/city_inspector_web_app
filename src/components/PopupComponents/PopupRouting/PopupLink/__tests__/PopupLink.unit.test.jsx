import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import PopupLink from '../PopupLink'

configure({ adapter: new Adapter() })

describe('<PopupLink />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let PopupLinkComponent
	const textContent = 'Example text'
	const to = 'FakePopup'
	const activeClassName = 'FakeCssClass'
	const className = 'FakeCssClass_2'

	test('should render correctly', () => {
		PopupLinkComponent = shallow(
			<PopupLink to={to} activeClassName={activeClassName} className={className}>
				{textContent}
			</PopupLink>,
		)

		expect(toJson(PopupLinkComponent)).toMatchSnapshot()
	})

	//TODO: tests related to context
})
