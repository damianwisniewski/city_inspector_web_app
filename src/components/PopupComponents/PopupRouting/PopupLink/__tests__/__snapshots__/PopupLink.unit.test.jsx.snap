// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupLink /> should render correctly 1`] = `
<button
  className="FakeCssClass_2 "
  data-to="FakePopup"
>
  Example text
</button>
`;
